technical/indicators/indicators.py

```diff
@@ -1353,3 +1353,38 @@ def tv_alma(
 
     alma = np.convolve(dataframe[field], weights[::-1], mode="valid")
     return Series(np.pad(alma, (length - 1, 0), mode="constant", constant_values=np.nan))
+
+
+def tv_trama(dataframe: DataFrame, length: int = 99, field="close") -> DataFrame:
+    """
+    Name : Tradingview "Trend Regularity Adaptive Moving Average"
+    Pinescript Author : LuxAlgo
+    Link :
+        tradingview.com/script/p8wGCPi6-Trend-Regularity-Adaptive-Moving-Average-LuxAlgo/
+
+    Args :
+        dataframe : Pandas Dataframe
+        timeperiod : Period of the indicator
+        source : Field to use for the calculation
+
+    Returns :
+        series : 'TRAMA' values
+    """
+
+    import talib.abstract as ta
+
+    df_len = len(dataframe)
+
+    hh = ta.MAX(dataframe["high"], length)
+    ll = ta.MIN(dataframe["low"], length)
+    hh_or_ll = np.where(np.diff(hh) > 0, 1, 0) + np.where(np.diff(ll) < 0, 1, 0)
+
+    tc = np.zeros(df_len)
+    tc[:-1] = np.nan_to_num(ta.SMA(hh_or_ll.astype(float), length) ** 2)
+
+    ama = np.zeros(df_len)
+    ama[0] = dataframe[field][0]
```

Comment:
it's not quite clear what you're trying to do - but this is quite error-prone as it assumes that an index with "0" exists.

This is not necessarily the case and not something we should rely on (a user could've moved the date column to the index, removed starting lines, ...).
If you really need the very first value of the list - please use `.iloc[0]` instead.

obviously the same applies to the below line.

You can test this pretty simply by passing `dataframe[1000:]` into the function (with a dataframe that has more than 1000 lines).

Reply:
Got it. Thanks for the help.